data/data/bootstrap/files/usr/local/bin/bootkube.sh.template

```diff
@@ -38,6 +38,20 @@ OPENSHIFT_HYPERKUBE_IMAGE=$(podman run --quiet --rm ${release} image hyperkube)
 
 CLUSTER_BOOTSTRAP_IMAGE=$(podman run --quiet --rm ${release} image cluster-bootstrap)
 
+# Verify at this point we have no images from docker.io or gcr.io
+repos=$(mktemp --suffix='bootkube')
+podman images --sort repository --format '{{"{{"}}.Repository {{"}}"}}' | sort -u > ${repos}
+if grep -E '(docker\.io|gcr\.io)/' ${repos}; then
+  echo 'Disallowed registries found!' 1>&2
+  exit 1
+fi
+
+# Now, as early as possible we replace the pause image and restart crio to use it, to ensure
+# that we're using the pause image from our payload just like the primary cluster.
+# Nothing should have created a pod yet.
+sed -i -e 's,pause_image *=.*,pause_image = "'${MACHINE_CONFIG_INFRA_IMAGE}'",' /etc/crio/crio.conf
+systemctl restart cri-o.service
```

Comment:
why is a restart required for CRI-O, are there some docs?

Reply:
While crio claims to support reload, currently `systemctl reload crio.service` sends `SIGHUP` and then apparently there's no signal handler, so that kills it...

Reply:
:man_facepalming:

Is there a CRI-O ticket for this?

Reply:
https://github.com/cri-o/cri-o/issues/2391

Reply:
Support for handling config changes (not all fields though) is being added to crio master. See https://github.com/cri-o/cri-o/pull/2349/files